aten/src/ATen/native/cuda/CUDALoops.cuh

```diff
@@ -151,9 +149,20 @@ static inline void launch_vectorized_kernel(
   constexpr auto io_size = calc_io_size<func_t>();
   int64_t grid = (N + io_block_work_size<io_size>() - 1) / io_block_work_size<io_size>();
   auto stream = at::cuda::getCurrentCUDAStream();
-  int vec_size = memory::can_vectorize_up_to<func_t>(data);
+  const uint16_t max_vec_size = memory::can_vectorize_elementwise_up_to<func_t>(data);
+  using cpp_type = typename function_traits<func_t>::result_type;
+  uint16_t vec_size = 16 / static_cast<uint16_t>(sizeof(cpp_type));
+  if(vec_size == 16) {
+    vec_size = 4;
```

Comment:
why is this vec_size set to 4 and not 8? Also, why is this conditional needed at all?

Reply:
This condition I used to avoid `vec8` for 1-byte data (`bool` and `uint8`) as some tests in `test/test_binary_ufuncs.py` were failing with mismatches.